plugin/nginx_plus_collectd.py

```diff
@@ -293,6 +300,17 @@ def configure(self, conf):
                 username = node.values[0]
             elif node.key == PASSWORD:
                 password = node.values[0]
+            elif node.key == API_VERSION:
+                err_msg = "{err}, please provide a valid positive integer value for the APIVersion"
+                try:
+                    api_version = int(node.values[0])
+
+                    if api_version < 0:
+                        raise ValueError("Invalid value found")
```

Comment:
We could include the invalid `api_version` to help flag the issue.

Reply:
This error message will be followed by the message `please provide a valid positive integer value for the APIVersion`, as any raised error in this block will be followed by the `err_msg` defined in that block.

Reply:
Right, but it doesn't include the actual provided value for `api_version`? adding this to the ValueError to be re-raised would be informative.

Reply:
Makes sense. I have updated the error message accordingly.